### PR TITLE
fix: make tenant SHOW TABLE STATUS rows consistent for system statement_info

### PR DESCRIPTION
**Pull request by robll-v1**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #10163

## What this PR does / why we need it:
Background / Root Cause
This PR fixes issue #10163: under a tenant account, SHOW TABLE STATUS for system.statement_info could show Rows = 0 (or otherwise inconsistent values) even when tenant-visible rows existed.

Root cause:

SHOW TABLE STATUS originally used mo_table_rows(db, tbl) for Rows.
For special system tables (for example system.statement_info), tenant queries are rewritten to run under the sys account with tenant-level filtering.
mo_table_rows does not always reflect that rewritten tenant-visible result set, so Rows could be inaccurate.
Solution
In handleShowTableStatus, add a targeted fallback path:

For non-sys accounts, and only for tables where ShouldSwitchToSysAccount(db, tbl) is true, run select count(*) on that table.
Use this count(*) result to override the Rows column in SHOW TABLE STATUS.
Keep existing behavior for all other tables (mo_table_rows/mo_table_size).
Route internal SQL execution through ExeSqlInBgSes (wrapper) to make stubbing in unit tests straightforward.
Test Coverage
Unit Tests ([mysql_cmd_executor_test.go](https://file+.vscode-resource.vscode-cdn.net/Users/moukeyu/.vscode/extensions/openai.chatgpt-0.4.73-darwin-x64/webview/#))
Added/covered boundary cases:

Fallback success path: mo_table_rows=0, count(*)>0, Rows is correctly overridden.
Non-special table: fallback is not triggered.
Sys account: fallback is not triggered.
Empty count(*) result: keep mo_table_rows.
count(*) execution error path.
count(*) decode (GetInt64) error path.
BVT (test/distributed/cases/dml/show/show7.sql/.result)
Added issue regression case with stabilization:

Tenant-session warm-up statements + wait (sleep) to avoid async log timing flakiness.
Pre-check count(*) > 0 on system.statement_info.
Validate SHOW TABLE STATUS ... LIKE 'statement_info' returns positive Rows (regex-based assertion).
Scope / Risk
Behavior change is limited to: non-sys account + special system tables requiring sys-account switch.
Adds one extra count(*) query only in that narrow path.
All other SHOW TABLE STATUS behavior remains unchanged.

### CODE REVIEW

**Review by heni02 — APPROVED** [review]

File: test/distributed/cases/dml/show/show7.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+drop account if exists show_table_status_acc_10163;
+create account show_table_status_acc_10163 admin_name = 'admin' identified by '111';
+
+-- @session:id=1&user=show_table_status_acc_10163:admin&password=111
+select 1;
+select 2;
+select 3;
+select sleep(15);
```
[the remark this answers is not in the document]
Comment (robll-v1) [reply test/distributed/cases/dml/show/show7.sql]: Good point. I replaced the unconditional `sleep(15)` with a conditional wait, so the case returns immediately when `system.statement_info` is already populated and only waits when needed. This keeps the regression test stable while avoiding a fixed 15-second delay on every run.
